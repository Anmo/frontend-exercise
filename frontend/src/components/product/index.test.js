import { fireEvent, render, screen } from '@testing-library/react';
import { useRecoilValue } from 'recoil';

import Product from './index';

jest.mock('../../store/product', () => ({
	productInfo: jest.fn()
}));

jest.mock('recoil', () => ({
	...jest.requireActual('recoil'),
	useRecoilValue: jest.fn()
}));

test('renders the normal Product component', () => {
	useRecoilValue.mockReturnValue({ name: 'test' });
	const productId = 'productId';

	const { container } = render(<Product productId={productId} />);

	expect(container.firstChild).toMatchSnapshot();
});

test('renders the normal Product component with buy message (with price)', () => {
	useRecoilValue.mockReturnValue({ name: 'test', price: 20 });
	const productId = 'productId';

	const { container } = render(<Product productId={productId} actionMessage />);

	expect(container.firstChild).toMatchSnapshot();
});

test('onClick should return productId', () => {
	useRecoilValue.mockReturnValue({ name: 'test' });
	const productId = 'productId';
	const onClick = jest.fn();

	render(<Product productId={productId} onClick={onClick} />);

	fireEvent.click(screen.getByRole('button'));

	expect(onClick).toBeCalledWith(productId);
});
