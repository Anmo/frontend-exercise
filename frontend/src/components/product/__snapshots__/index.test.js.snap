// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the normal Product component 1`] = `
<div
  class="MuiPaper-root MuiCard-root container MuiPaper-elevation1 MuiPaper-rounded"
  tabindex="1"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root action"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          test
        </span>
      </div>
    </div>
    <div
      class="MuiCardMedia-root media"
      style="background-image: url(/icons/productId.svg);"
    />
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`renders the normal Product component with buy message (with price) 1`] = `
<div
  class="MuiPaper-root MuiCard-root container MuiPaper-elevation1 MuiPaper-rounded"
  tabindex="1"
>
  <button
    class="MuiButtonBase-root MuiCardActionArea-root action"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          test
        </span>
      </div>
    </div>
    <div
      class="MuiCardMedia-root media"
      style="background-image: url(/icons/productId.svg);"
    />
    <p
      class="MuiTypography-root MuiTypography-body1"
    >
      Click to buy (
      20
      )
    </p>
    <span
      class="MuiCardActionArea-focusHighlight"
    />
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
